authenticators/components/org.wso2.carbon.identity.sample.local.authenticator/README.md

```diff
@@ -0,0 +1,81 @@
+
+<br>
+
+**NOTE:** 
+The customization and configuration details of this guide and the sample are given based on the **WSO2 Identity Server 5.11.0** 
+
+<br>
+
+## Implement the custom local authenticator
+
+1. Create a maven project for the custom authenticator. You can refer to the [pom.xml](./pom.xml) file used to implement the sample local authenticator that is discussed in this guide.
+2. The authenticator should be written as an OSGi service and deployed in the WSO2 Identity Server. Therefore, we need to have a [service component class](src/main/java/org/wso2/carbon/identity/sample/local/authenticator/internal/SampleLocalAuthenticatorServiceComponent.java) in our sample authenticator to register it as a local authenticator.
+3. The [custom local authenticator](src/main/java/org/wso2/carbon/identity/sample/local/authenticator/SampleLocalAuthenticator.java) should be written by extending the [AbstractApplicationAuthenticator](https://github.com/wso2/carbon-identity-framework/blob/v5.18.187/components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/AbstractApplicationAuthenticator.java) class and implementing the [LocalApplicationAuthenticator](https://github.com/wso2/carbon-identity-framework/blob/v5.18.187/components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/LocalApplicationAuthenticator.java) class.
+
+### Methods that need to be implemented,
+
+**getFriendlyName():** returns the name you want to display for your custom authenticator.
+* Our sample custom local authenticator is named as sample-local-authenticator
+
+**getname():** get the name of the authenticator.
+
+**canHandle():** checks whether the authentication request is valid, according to the custom authenticator’s requirements. The user will be authenticated if the method returns ‘true’. This method also checks whether the authentication or logout request can be handled by the authenticator.
+* For example, you can check if the username and password is ‘not null’ in the canHandle method. if that succeeds, the authentication flow will continue (to have the user authenticated).
+
+**processAuthenticationResponse():** Implementation of custom authentication logic happens inside this method.
+* In our example, we use the following API where the user claim used as the username is passed along with the value and authenticate the user. Since we are using the telephone number as the login identifier, the prefferedUsernameClaim is http://wso2.org/claims/telephone.
```

Comment:
```suggestion
* In our example, we use the following API where the user claim used as the username is passed along with the value and authenticate the user. Since we are using the telephone number as the login identifier, the preferredUsernameClaim is http://wso2.org/claims/telephone.
```